### PR TITLE
introduce envsubst for config.js, update deps

### PR DESCRIPTION
**Pull request by khassel**
This is the implemenation of envsubst discussed in #1756 

Documentation update will follow after merge.

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -56,13 +57,62 @@ function App() {
 	 *
 	 * @param {Function} callback Function to be called after loading the config
 	 */
-	function loadConfig(callback) {
+	async function loadConfig(callback) {
```
Comment (rejas) [inline js/app.js:60]: I would like to see the loadConfig being asynced and also being freed from the callback. Have you tried that?

**Review by rejas — CHANGES_REQUESTED** [review]
Can you add an example(maybe just with a comment in the sample.config so that people know how to use it?
Would it be possible to add a test for this?
And of course the async question if we convert the whole loadCOnfig mthod to async without callback?

File: .eslintignore [hunk 2]
```diff
@@ -1 +1,2 @@
 modules/default/calendar/vendor/*
+tests/configs/port_variable.js
```
Comment (rejas) [inline .eslintignore:2]: why are these two ignores necessary? locally nothing complains about them....

Comment (khassel) [reply .eslintignore:2]: you are right, will remove them. Had first `port_variable.js` (forgot `.template` at the end) and with `.js` the tests failed.

**Review by rejas — APPROVED** [review]

**Pull request merged**